asg-instance-refresh.go

```diff
@@ -1,98 +1,63 @@
 package main
 
 import (
+	"context"
+	"errors"
 	"flag"
 	"fmt"
 	"os"
-	"time"
-
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"os/signal"
+	"strings"
 )
 
-var (
-	region     = flag.String("region", "us-east-1", "The AWS region. Default to us-east-1.")
-	warmup     = flag.Int64("warmup", -1, `How long the new instances need to warm up. Default to the value for the health check grace period defined for the group.`)
-	minHealthy = flag.Int64("minhealthy", 100, `The minimum percentage capacity to retain during the recycling. Default to 100%.`)
-	wait       = flag.Bool("wait", false, "Wait until the refresh is complete to return.")
-)
+var ErrStrategyFailed = errors.New("strategy failed")
 
 func main() {
+	region := flag.String("region", "us-east-1", "The AWS region. Default to us-east-1.")
```

Comment:
Minor/comment: if like me you get sick of typing `*region` everywhere you can use `var region string; flag.StringVar(&region, ....)`.

Not suggesting you make this change as part of this PR just a TIL.

Reply:
if I get that sick of it, I'll move to `cobra` or something like that ;)